### DISCUSSION TITLE
Deploying to Azure

### DISCUSSION THREAD

**Discussion by SkyBlade1978** [itzg/docker-minecraft-server#848] [Q&A]
Hi,

So I now have my container working in desktop docker, next step to pop it online so people can play it, and I would like to use azure.

I wonder is there a step-by-step for this, I understand I'll need to create an azure file share for the volume, at the moment I'm stuck on the environment variables tab of creating a container as I am not sure the correct way to transpose my docker terminal command I used to create the container into the UI..

For example I have;

docker run -d -it -v mcvol:/data -e TYPE=FORGE -p 25565:25565 -e EULA=TRUE -e VERSION=1.10.2 -e JVM_DD_OPTS=fml.queryResult:confirm --name mc itzg/minecraft-server:java8

And in the azure ui for the variables I have two fields, key and value, if I need to pass in -it is that a key or a value?

![image](https://user-images.githubusercontent.com/22159126/116368336-833dc400-a808-11eb-9bb5-6472dea3bdb8.png)

**Comment by rampd2** [comment]
Why not use a [Azure VM](https://azure.microsoft.com/services/virtual-machines/) and then install docker etc. on it?

**Reply by itzg** [reply]
I agree with @DanielRamp that running docker yourself on a VM is possibly the best way to go. Otherwise, the docs for compose support look promising

https://docs.microsoft.com/en-us/azure/container-instances/tutorial-docker-compose

**Comment by DavidRBailey** [comment]
Some potential resources.

https://markheath.net/post/minecraft-server-azure-container-instances

https://github.com/geekzter/azure-minecraft-docker
